packages/mgt/src/providers/MsalProvider.ts

```diff
@@ -330,38 +356,59 @@ export class MsalProvider extends IProvider {
     return false;
   }
 
-  private initProvider(config: MsalConfig) {
+  private initProvider(config: MsalConfig | MsalUserAgentApplicationConfig) {
     this.scopes = typeof config.scopes !== 'undefined' ? config.scopes : ['user.read'];
     this._loginType = typeof config.loginType !== 'undefined' ? config.loginType : LoginType.Redirect;
     this._loginHint = config.loginHint;
 
-    if (config.clientId) {
-      const msalConfig: Configuration = config.options || { auth: { clientId: config.clientId } };
+    let userAgentApplication: UserAgentApplication;
+    let clientId: string;
 
-      msalConfig.auth.clientId = config.clientId;
-      msalConfig.cache = msalConfig.cache || {};
-      msalConfig.cache.cacheLocation = msalConfig.cache.cacheLocation || 'localStorage';
-      msalConfig.cache.storeAuthStateInCookie = msalConfig.cache.storeAuthStateInCookie || true;
+    if ('clientId' in config) {
+      if (config.clientId) {
+        const msalConfig: Configuration = config.options || { auth: { clientId: config.clientId } };
 
-      if (config.authority) {
-        msalConfig.auth.authority = config.authority;
-      }
+        msalConfig.auth.clientId = config.clientId;
+        msalConfig.cache = msalConfig.cache || {};
+        msalConfig.cache.cacheLocation = msalConfig.cache.cacheLocation || 'localStorage';
+        msalConfig.cache.storeAuthStateInCookie = msalConfig.cache.storeAuthStateInCookie || true;
 
-      if (config.redirectUri) {
-        msalConfig.auth.redirectUri = config.redirectUri;
-      }
+        if (config.authority) {
+          msalConfig.auth.authority = config.authority;
+        }
+
+        if (config.redirectUri) {
+          msalConfig.auth.redirectUri = config.redirectUri;
+        }
 
-      this.clientId = config.clientId;
+        clientId = config.clientId;
 
-      this._userAgentApplication = new UserAgentApplication(msalConfig);
-      this._userAgentApplication.handleRedirectCallback(
-        response => this.tokenReceivedCallback(response),
-        (error, state) => this.errorReceivedCallback(error, state)
-      );
+        userAgentApplication = new UserAgentApplication(msalConfig);
+      } else {
+        throw new Error('clientId must be provided');
+      }
+    } else if ('userAgentApplication' in config) {
+      if (config.userAgentApplication instanceof UserAgentApplication) {
```

Comment:
```suggestion
      if (config.userAgentApplication) {
```

@shweaver-MSFT How would you feel about this? I got it all working before I opened this, but as I was getting #582 together, I had a few runtime errors (i.e. TypeScript let through) with package version conflicts: pass in a perfectly good `MsalService`, and it this would return false.

I'm 99% sure it was just because I was using `npm link`, and it was sourcing from multiple `node_modules` directories. I did manage to get it working, and I'm fairly certain it would all be okay once it's published on npm. But at the same time, I wonder whether this check is worth it, or if it will just add unnecessary risk that could be impossible to fix in a caller's code.

I don't feel strongly. I've definitely gotten this working as-is, and I feel like this is a native feature that should just work. But I'm also not sure I see much real advantage, in retrospect. It's not like we check if `config.clientId` is a string.